Cargo.toml

```diff
@@ -45,4 +45,4 @@ tokio-stream = "0.1.1"
 tracing = { version = "0.1", default-features = false }
 tracing-core = { version = "0.1", default-features = false }
 tracing-subscriber = { version = "0.3", default-features = false }
-url = { version = "2.2", default-features = false }
+url = { version = "=2.5.0", default-features = false } #pinning the version supporting rustc 1.65
```

Comment:
Why need ping this version to support the given rustc?

Reply:
If not pinned, the higher version ` 2.5.1` will be used, which requires rust `1.67`. The previous configured version `2.2` did not have any impact since `url` is an indirect dependency, and thus, the rules defined by the direct dependency that uses it are applied.